app/src/main/java/com/example/myapplication/CredentialsManager.kt

```diff
@@ -41,16 +41,14 @@ class CredentialsManager {
     }
 
     fun register(fullName: String, phoneNumber: String, email: String, password: String): Boolean {
-        if (fullName.isEmpty() || phoneNumber.isEmpty() || !isEmailValid(email) || !isPasswordValid(
-                password
-            )
-        ) return false
         if (credentials.containsKey(email.lowercase())) {
             return false
         } else {
-            credentials[email.lowercase()] = password
```

Comment:
I don't understand why you removed the functionality of saving a new account in the credentials map

Reply:
Because if you were to enter an email and keep every other input text empty, the email would still be saved in the system.

Reply:
No because there's a check at the start of the function that controls that every text input is valid (and that also includes the not-empty clause of the email).

Reply:
Yes I see, there probably should be a way to have the correct error message for the emailInputLayout with this version of the function.